src/android/MainActivity.java

```diff
@@ -0,0 +1,656 @@
+            recorder.prepare();
+
+        } catch (IOException e) {
+            e.printStackTrace();
+
+        }
+    }
+
+
+    private void startBackgroundThread() {
+        backgroundThread = new HandlerThread("CameraBackground");
+        backgroundThread.start();
+        backgroundHandler = new Handler(backgroundThread.getLooper());
+    }
+
+    public void cancelActivity() {
+
+        if (isRecordingVideo) {
+            File file = new File(lastRecordedFileUrl);
+            file.delete();
+        }
+        finish();
+    }
+    public void startRecordingVideo() {
+        if (isRecordingVideo) {
+            return;
+        }
+        if (null == cameraDevice || !textureView.isAvailable() || null == previewsize) {
+            return;
+        }
+        try {
+            closePreviewSession();
+            prepareMediaRecorder();
+            SurfaceTexture texture = textureView.getSurfaceTexture();
+            assert texture != null;
+            texture.setDefaultBufferSize(previewsize.getWidth(), previewsize.getHeight());
+            previewBuilder = cameraDevice.createCaptureRequest(CameraDevice.TEMPLATE_RECORD);
+            List<Surface> surfaces = new ArrayList<>();
+
+            Surface previewSurface = new Surface(texture);
+            surfaces.add(previewSurface);
+            previewBuilder.addTarget(previewSurface);
+            //MediaRecorder setup for surface
+            Surface recorderSurface = recorder.getSurface();
+            surfaces.add(recorderSurface);
+            previewBuilder.addTarget(recorderSurface);
+            // Start a capture session
+            cameraDevice.createCaptureSession(surfaces, new CameraCaptureSession.StateCallback() {
+                @Override
+                public void onConfigured( CameraCaptureSession cameraCaptureSession) {
+                    previewSession = cameraCaptureSession;
+                    updatePreview();
+
+                    MainActivity.this.runOnUiThread(() -> {
+                        isRecordingVideo = true;
+                        lockDeviceRotation(true);
+                        // Start recording
+                        recorder.start();
+                        stopRecordingButton.show();
+                        recordVideoButton.hide();
+                        stopRecordingButton.setAlpha(0.66f);
+
+
+                        myTimer = new Timer();
+                        myTimer.schedule(new TimerTask() {
+                            @Override
+                            public void run() {
+                                TimerMethod();
+                            }
+
+                        }, 0, 1000);
+                    });
+
+                }
+                @Override
+                public void onConfigureFailed( CameraCaptureSession cameraCaptureSession) {
+                    Log.e("", "onConfigureFailed: Failed");
+                }
+            }, backgroundHandler);
+        } catch (CameraAccessException e) {
```

Comment:
If camera access is not accepted, what should happen? Probably not a problem to be solved right now

Reply:
to be solved later